test/integration/challenges.test.js

```diff
@@ -0,0 +1,111 @@
+const chai = require('chai')
+const { expect } = chai
+const chaiHttp = require('chai-http')
+
+const app = require('../../server')
+const authService = require('../../services/authService')
+const addUser = require('../utils/addUser')
+const cleanDb = require('../utils/cleanDb')
+
+const config = require('config')
+const cookieName = config.get('userToken.cookieName')
+
+chai.use(chaiHttp)
+
+describe('Challenges', function () {
+  let jwt
+
+  beforeEach(async function () {
+    const userId = await addUser()
+    jwt = authService.generateAuthToken({ userId })
+  })
+
+  afterEach(async function () {
+    await cleanDb()
+  })
+
+  describe('GET /challenges', function () {
+    it('Should return the available challenges', function (done) {
+      chai
+        .request(app)
+        .get('/challenges')
+        .end((err, res) => {
+          if (err) {
+            return done(err)
+          }
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Challenges returned successfully!')
+          expect(res.body.challenges).to.be.a('array')
+
+          return done()
+        })
+    })
+    it('Should return a not found message if the challenges is not found', function (done) {
+      chai
+        .request(app)
+        .get('/challenges')
+        .end((err, res) => {
+          if (err) {
+            return done()
+          }
+          expect(res).to.have.status(404)
+          expect(res.body.message).to.be.equal('No Challenges found')
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /challenges', function () {
+    it('Should create a new challenge', function (done) {
+      chai
+        .request(app)
+        .post('/challenges')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send()
```

Comment:
We need to pass the requestBody for `/challenges` it can not be empty. You can see similar example [here](https://github.com/Real-Dev-Squad/website-backend/blob/5ec05e86695a2e717030156e91f779f45c611adf/test/integration/tasks.test.js?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L85)

Reply:
okay